src/algebra/ring/aut.lean

```diff
@@ -60,4 +61,26 @@ by refine_struct { to_fun := ring_equiv.to_mul_equiv }; intros; refl
 def to_perm : ring_aut R →* equiv.perm R :=
 by refine_struct { to_fun := ring_equiv.to_equiv }; intros; refl
 
+/-- The tautological action by the group of automorphism of a ring `R` on `R`.-/
+instance apply_mul_semiring_action {R : Type*} [ring R] : mul_semiring_action (ring_aut R) R :=
+{ smul := ($),
+  smul_zero := ring_equiv.map_zero,
+  smul_add := ring_equiv.map_add,
+  smul_one := ring_equiv.map_one,
+  smul_mul := ring_equiv.map_mul,
+  one_smul := λ _, rfl,
+  mul_smul := λ _ _ _, rfl }
+
```

Comment:
Can you add the analogs of these lemmas too?

https://github.com/leanprover-community/mathlib/blob/2a528d11813a4452202e79a3c2d82fd7db2bc6e8/src/algebra/algebra/basic.lean#L1273-L1276

Reply:
I just added them.